### PR TITLE
[SPARK-41372][CONNECT][PYTHON] Implement DataFrame TempView

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Implement DataFrame TempView (which is createTemView and createOrReplaceTempView). This is a session local temp view which is different from the global temp view.

### Why are the changes needed?

API coverage.

### Does this PR introduce _any_ user-facing change?

NO
### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
cc @zhengruifeng @HyukjinKwon

**Comment by amaliujia** [comment]
also cc @xinrong-meng

File: python/pyspark/sql/connect/dataframe.py [hunk 1]
```diff
@@ -1554,6 +1554,48 @@ def explain(
...
 
+    def createTempView(self, name: str) -> None:
+        """Creates a local temporary view with this :class:`DataFrame`.
+
+        The lifetime of this temporary table is tied to the :class:`SparkSession`
+        that was used to create this :class:`DataFrame`.
+        throws :class:`TempTableAlreadyExistsException`, if the view name already exists in the
+        catalog.
+
+        .. versionadded:: 2.0.0
```
Comment (xinrong-meng) [inline python/pyspark/sql/connect/dataframe.py:1565]: 3.4.0?

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -1554,6 +1554,48 @@ def explain(
...
+
+        Parameters
+        ----------
+        name : str
+            Name of the view.
+        """
```
Comment (xinrong-meng) [inline python/pyspark/sql/connect/dataframe.py:1571]: nit:
```
        Returns
        -------
        None
```
for consistency

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 3]
```diff
@@ -498,11 +498,23 @@ def test_create_global_temp_view(self):
...
+    def test_create_session_local_temp_view(self):
+        # SPARK-41372: test session local temp view creation.
+        with self.tempView("view_local_temp"):
+            self.connect.sql("SELECT 1 AS X").createTempView("view_local_temp")
+            self.assertEqual(self.connect.sql("SELECT * FROM view_local_temp").count(), 1)
+            self.connect.sql("SELECT 1 AS X LIMIT 0").createOrReplaceTempView("view_local_temp")
+            self.assertEqual(self.connect.sql("SELECT * FROM view_local_temp").count(), 0)
+
+            # Test when creating a view which is already exists but
+            with self.assertRaises(grpc.RpcError):
```
Comment (xinrong-meng) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:515]: Out of curiosity, don't we have a way to parse `TempTableAlreadyExistsException` out of `grpc.RpcError`?

Comment (amaliujia) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:515]: We don't have it now. This is a future work that we need to finish.

File: python/pyspark/sql/connect/dataframe.py [hunk 4]
```diff
@@ -1554,6 +1554,52 @@ def explain(
...
+
+        Parameters
+        ----------
+        name : str
+            Name of the view.
+
+        Returns
+        -------
+        None
+        """
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/dataframe.py:1575]:
```suggestion
        """
```

File: python/pyspark/sql/connect/dataframe.py [hunk 5]
```diff
@@ -1554,6 +1554,52 @@ def explain(
...
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        name : str
+            Name of the view.
+
+        Returns
+        -------
+        None
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/dataframe.py:1596]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Review by grundprinzip — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 3 above)
Comment (xinrong-meng) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:515]: Sounds good, thanks.

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
Merged to master, thanks!

**Pull request closed without merging**